lib/SILOptimizer/Mandatory/DiagnoseStaticExclusivity.cpp

```diff
@@ -162,8 +162,32 @@ class AccessedStorage {
   }
 };
 
-/// Models the in-progress accesses for a single storage location.
-class AccessInfo {
+/// Records an access to an address and the single subpath of projections
+/// that was performed on the address, if such a single subpath exists.
+class RecordedAccess {
+private:
+  BeginAccessInst *Inst;
+  ProjectionPath SubPath;
```

Comment:
We discussed this in person but in the interest of public information,
it is unfortunate that we have a fully general looking utility called
"ProjectionPath" that is usually not what a pass should use to
represent projection paths. When representing access paths across a
multitude of accesses, a prefix trie should be used with each path
represented by a single pointer that identifies a unique integer
sequence along with any pass-specific data.

This approach should be both simpler and much more scalable. The
Load/Store optimizer regrettably made the mistake of using
ProjectionPath. It is quite expensive and not scalable.
We definitely should not build ProjectionPaths at -Onone.